src/Dfe.PlanTech.Application/Users/Commands/RecordUserSigninCommand.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Dfe.PlanTech.Application.Persistence.Interfaces;
+using Dfe.PlanTech.Application.Users.Interfaces;
+using Dfe.PlanTech.Application.Users.Queries;
+using Dfe.PlanTech.Domain.SignIn.Models;
+using Dfe.PlanTech.Domain.Users.Models;
+
+namespace Dfe.PlanTech.Application.Users.Commands;
+
+public class RecordUserSignInCommand : IRecordUserSignInCommand
+{
+    private readonly IUsersDbContext _db;
+
+    public RecordUserSignInCommand(IUsersDbContext db)
+    {
+        _db = db;
+    }
+
+    public async Task<int> RecordSignIn(RecordUserSignInDto recordUserSignInDto)
+    {
+        //Check user exists already
+        var getUserIdQuery = new GetUserIdQuery(_db);
+
+        var existingUserId = await getUserIdQuery.GetUserId(recordUserSignInDto.DfeSignInRef);
+
+        if (existingUserId == null)
+        {
+            var CreateUserCommand = new CreateUserCommand(_db);
+            await CreateUserCommand.CreateUser(recordUserSignInDto);
+            existingUserId = await getUserIdQuery.GetUserId(recordUserSignInDto.DfeSignInRef);
+        }
+
+        var signInId = await AddSignInDetails(MapToSignIn(existingUserId));
+        return signInId;
+    }
+
+    private static Domain.SignIn.Models.SignIn MapToSignIn(int? userId)
+    {
+        return new Domain.SignIn.Models.SignIn
+        {
+            UserId = Convert.ToUInt16(userId),
+            EstablishmentId = 1, //Replace with value
```

Comment:
EstablishmentId = establishmentId == null ? 1 : establishmentId

Given we will assume we'll get establishmentId from a user from the DSI service

Reply:
> EstablishmentId = establishmentId == null ? 1 : establishmentId
> 
> Given we will assume we'll get establishmentId from a user from the DSI service

Establishment can't be set to null.  Originally I had this and it was giving me an error.  So I defaulted it to 1 for the time being.

Reply:
Changed to have a default argument on the method.